src/backends/disk.go

```diff
@@ -74,8 +83,59 @@ func (db *DiskBackend) Save(data []byte) error {
 	return nil
 }
 
+func (f fileInfoSlice) Len() int {
+	return len(f)
+}
+
+func (f fileInfoSlice) Less(i, j int) bool {
+	return f[i].ModTime().Before(f[j].ModTime())
+}
+
+func (f fileInfoSlice) Swap(i, j int) {
+	f[i], f[j] = f[j], f[i]
+}
+
+func (db *DiskBackend) cleanOldBackups(max int) error {
+	files, err := ioutil.ReadDir(db.backupConfig.DirPath)
+	if err != nil {
+		return err
+	}
+
+	filesSorted := make(fileInfoSlice, 0, len(files))
+	for _, f := range files {
+		filesSorted = append(filesSorted, f)
+	}
+	sort.Sort(filesSorted)
+	max = min(max, len(filesSorted))
+
+	for _, f := range filesSorted[:len(filesSorted)-max] {
+		p := fmt.Sprintf("%s/%s", db.backupConfig.DirPath, f.Name())
+		if err := os.Remove(p); err != nil {
+			fmt.Println("Error", err.Error())
+		}
+	}
+
+	return nil
+}
+
+func min(a, b int) int {
+	if a <= b {
+		return a
+	}
+	return b
+}
+
 func (db *DiskBackend) Backup() error {
-	backupDir := fmt.Sprintf(defaultBackupDir, homeDir, defaultSafeDirName)
+	if db.backupConfig.MaxFiles == 0 {
+		// Keep no backups
+		db.cleanOldBackups(0)
+		return &errors.BackupDisabled{}
+	} else if db.backupConfig.MaxFiles > 0 {
+		// Subtract one as we are about to create another backup
+		db.cleanOldBackups(db.backupConfig.MaxFiles - 1)
```

Comment:
It's unexpected behavior, and as a user that is not something I want from my password manager.

If the error is not fatal, which I agree it is not, we should at least surface something to the user , simlilar to https://github.com/bndw/pick/pull/68#discussion-diff-79317027R114, since there are multiple operations performed.